utils/clean-up-job.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: buildkite-agent-stack-k8s-job-cleanup
+  namespace: buildkite
+spec:
+  schedule: "*/5 * * * *"  # Run every 5 minutes
+  successfulJobsHistoryLimit: 1
+  failedJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      backoffLimit: 1
+      template:
+        spec:
+          serviceAccountName: agent-stack-k8s-controller
+          containers:
+          - name: buildkite-agent-stack-k8s-job-cleanup
+            image: bitnami/kubectl:latest
+            imagePullPolicy: IfNotPresent
+            env:
+            - name: CLEANUP_K8S_NAMESPACE
+              value: "buildkite" # Default agent-stack-k8s namespace
```

Comment:
This is also tricky if the user is using a different namespace. Is there a way to infer this?